File: converter/terrain_converter/tilejson.py
"""Generate TileJSON metadata for terrain tiles."""

from __future__ import annotations

import json
from pathlib import Path

from .bbox import Bounds


def build_tilejson(bounds: Bounds, min_zoom: int, max_zoom: int, tiles_url: str) -> dict[str, object]:
    center_lon, center_lat = bounds.center()
    return {
        "tilejson": "3.0.0",
        "name": "terrain-dem",
        "type": "raster-dem",
        "scheme": "xyz",
        "encoding": "mapbox",
        "format": "png",
        "tileSize": 256,
        "tiles": [tiles_url],
        "bounds": [bounds.west, bounds.south, bounds.east, bounds.north],
        "center": [center_lon, center_lat, min_zoom],
        "minzoom": min_zoom,
        "maxzoom": max_zoom,
    }


def write_tilejson(path: str | Path, tilejson: dict[str, object]) -> None:
    path = Path(path)
    path.parent.mkdir(parents=True, exist_ok=True)
    path.write_text(json.dumps(tilejson, indent=2) + "\n", encoding="utf-8")

File: kotlin/terrain-core/src/jvmMain/kotlin/com/terrainconverter/core/TileJson.kt
package com.terrainconverter.core

import java.nio.file.Files
import java.nio.file.Path

fun buildTileJson(
    bounds: Bounds,
    minZoom: Int,
    maxZoom: Int,
    tilesUrl: String,
    name: String = "terrain-dem",
    scheme: String = "xyz",
    encoding: String = "mapbox",
    tileSize: Int = 256,
): Map<String, Any?> {
    val (centerLon, centerLat) = bounds.center()
    return jsonObject(
        "tilejson" to "3.0.0",
        "name" to name,
        "type" to "raster-dem",
        "scheme" to scheme,
        "encoding" to encoding,
        "format" to "png",
        "tileSize" to tileSize,
        "tiles" to listOf(tilesUrl),
        "bounds" to listOf(bounds.west, bounds.south, bounds.east, bounds.north),
        "center" to listOf(centerLon, centerLat, minZoom),
        "minzoom" to minZoom,
        "maxzoom" to maxZoom,
    )
}

fun writeTileJson(path: Path, tileJson: Map<String, Any?>) {
    path.parent?.let { Files.createDirectories(it) }
    Files.writeString(path, renderPrettyJson(tileJson))
}
